xbmc/guilib/GUIControlFactory.cpp

```diff
@@ -126,7 +125,7 @@ std::string CGUIControlFactory::TranslateControlType(CGUIControl::GUICONTROLTYPE
   return "";
 }
 
-CGUIControlFactory::CGUIControlFactory(void) = default;
+CGUIControlFactory::CGUIControlFactory(GUIResourceProviderPtr provider) = default;
```

Comment:
error: `CGUIControlFactory::CGUIControlFactory(GUIResourceProviderPtr)` cannot be defaulted